overlord/ifacestate/helpers.go

```diff
@@ -269,12 +269,22 @@ func (m *InterfaceManager) reloadConnections(snapName string) ([]string, error)
 			continue
 		}
 
+		plugInfo := m.repo.Plug(connRef.PlugRef.Snap, connRef.PlugRef.Name)
+		slotInfo := m.repo.Slot(connRef.SlotRef.Snap, connRef.SlotRef.Name)
+
+		// After snap refresh a plug/slot may no longer be present. Connection should be removed from
+		// the state as long as it's an auto-connect and wasn't disconnected manually; note that undesired
+		// flag is taken care of at the beginning of the loop.
+		if (plugInfo == nil || slotInfo == nil) && connState.Auto && !connState.ByGadget && connState.Interface != "core-support" {
```

Comment:
no strong opinion but would unifying the two ifs with comments be clearer ?

```
// the connection refers to slot or plug that don't exist anymore
if plugInfo == nil || slotInfo == nil {
     // the connection was an auto-connect etc we can  remove it
     if connState.Auto .... {
         delete ...
     }
     //  we keep it in case there is a revert, this might happen so we silently ignore etc
     continue
}
```
^ to be clear comments are just sketched